packages/controller/lib/setup/setupRepo.js

```diff
@@ -227,154 +219,117 @@ function Repo(options) {
             }
         });
 
-        objects.getObjectView('system', 'instance', {startkey: 'system.adapter.admin', endkey: 'system.adapter.admin\u9999'}, null, (err, objs) => {
-            const instances = [];
-
-            err && console.error(err);
+        const objs = await objects.getObjectViewAsync('system', 'instance', {startkey: 'system.adapter.admin', endkey: 'system.adapter.admin\u9999'});
 
-            if (objs && objs.rows && objs.rows.length) {
-                for (let t = 0; t < objs.rows.length; t++) {
-                    if (objs.rows[t] && objs.rows[t].value && objs.rows[t].value.common && objs.rows[t].value.type === 'instance') {
-                        instances.push(objs.rows[t].id);
-                    }
+        if (objs && objs.rows && objs.rows.length) {
+            const listStr = list.join(', ');
+            for (let t = 0; t < objs.rows.length; t++) {
+                if (objs.rows[t] && objs.rows[t].value && objs.rows[t].value.common && objs.rows[t].value.type === 'instance') {
+                    await states.setStateAsync(objs.rows[t].id + '.info.updatesNumber', list.length, true);
+                    await states.setStateAsync(objs.rows[t].id + '.info.updatesList', listStr, true);
                 }
             }
-
-            writeUpdateInfo(instances, list.length, list.join(', '), callback);
-        });
+        }
     }
 
-    this.showRepoStatus = function (callback) {
-        objects.getObject('system.repositories', (err, obj) => {
-            if (err || !obj) {
-                console.error('Cannot get list: ' + err);
-                callback(EXIT_CODES.CANNOT_GET_REPO_LIST);
-            } else {
-                if (obj.native.repositories) {
-                    Object.keys(obj.native.repositories).forEach(r =>
-                        console.log(`${r.padEnd(14)}: ${obj.native.repositories[r].link}`));
-
-                    objects.getObject('system.config', (err, obj) => {
-                        if (obj && obj.common) {
-                            let activeRepo = obj.common.activeRepo;
-                            if (typeof activeRepo === 'string') {
-                                activeRepo = [activeRepo];
-                            }
-                            console.log(`\nActive repo(s): ${activeRepo.join(', ')}`);
-                        }
-                        callback();
-                    });
-                } else {
-                    console.error('Cannot get list: ' + err);
-                    callback(EXIT_CODES.CANNOT_GET_REPO_LIST);
+    this.showRepoStatus = async function () {
```

Comment:
Really with "function"? Also cases below. Yes, no "this" is used but still :-)

Reply:
this still work with such a construction. And it is all old stuff, that I did't change